src/AtmosphereModels/atmosphere_model.jl

```diff
@@ -5,7 +5,7 @@ using Oceananigans: AbstractModel, Center, CenterField, Clock, Field,
 using Oceananigans.Advection: adapt_advection_order
 using Oceananigans.AbstractOperations: @at
 using Oceananigans.BoundaryConditions: FieldBoundaryConditions, regularize_field_boundary_conditions
-using Oceananigans.Diagnostics: NaNChecker
+using Oceananigans.Diagnostics: Diagnostics, NaNChecker
```

Comment:
Ah, there's already a submodule called `Diagnostics` in `AtmosphereModels`!  We have two options:

* do `using Oceananigans: Oceananigans`, and then define the method `Oceananigans.Diagnostics.default_nan_checker(...)`, or
* do `using Oceananigans.Diagnostics: Diagnostics as OceananigansDiagnostics, NaNChecker` (or some other alias), and then define the method `OceananigansDiagnostics.default_nan_checker(...)`